modules/xfeatures2d/src/msd.cpp

```diff
@@ -0,0 +1,791 @@
+        {
+
+            const float deriv_scale = 0.5f;
+            int width_s = m_scaleSpace[scale].cols;
+            //const float second_deriv_scale = 1.0f;
+            const float cross_deriv_scale = 0.25f;
+
+            cv::Vec2f dD((saliency[scale][j * width_s + i + 1] - saliency[scale][j * width_s + i - 1]) * deriv_scale,
+                    (saliency[scale][(j + 1) * width_s + i] - saliency[scale][(j - 1) * width_s + i]) * deriv_scale);
+
+            float cc = saliency[scale][j * width_s + i] * 2;
+            float dxx = (saliency[scale][j * width_s + i + 1] + saliency[scale][j * width_s + i - 1] - cc);// * second_deriv_scale;
+            float dyy = (saliency[scale][(j + 1) * width_s + i] + saliency[scale][(j - 1) * width_s + i] - cc);// * second_deriv_scale;
+            float dxy = (saliency[scale][(j + 1) * width_s + i + 1] - saliency[scale][(j + 1) * width_s + i - 1] -
+                    saliency[scale][(j - 1) * width_s + i + 1] + saliency[scale][(j - 1) * width_s + i - 1]) * cross_deriv_scale;
+            
+            cv::Matx22f H(dxx, dxy, dxy, dyy);
+
+            cv::Vec2f X = H.solve(dD, cv::DECOMP_LU);
+
+            float xr = -X[1];
+            float xc = -X[0];
+
+            if (std::abs(xr) > 5 || std::abs(xc) > 5)
+                return false;
+
+            if (scale == 0)
+            {
+                p_res.x = i + xc + 0.5f;
+                p_res.y = j + xr + 0.5f;
+            } else
+            {
+                float effectiveScaleFactor = std::pow(m_scale_factor, scale);
+                p_res.x = (i + xc + 0.5f) * effectiveScaleFactor;
+                p_res.y = (j + xr + 0.5f) * effectiveScaleFactor;
+
+                p_res.x -= 0.5f;
+                p_res.y -= 0.5f;
+
+                if (p_res.x < 0 || p_res.x >= m_scaleSpace[0].cols || p_res.y < 0 || p_res.y >= m_scaleSpace[0].rows)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        void MSDDetector_Impl::contextualSelfDissimilarity(cv::Mat &img, int xmin, int xmax, float* saliency)
+        {
+            int r_s = m_patch_radius;
+            int r_b = m_search_area_radius;
+            int k = m_kNN;
+
+            int w = img.cols;
+            int h = img.rows;
+
+            int side_s = 2 * r_s + 1;
+            int side_b = 2 * r_b + 1;
+            int border = r_s + r_b;
+            int temp;
+            int den = side_s * side_s * k;
+
+            std::vector<int> minVals(k);
+            int *acc = new int[side_b * side_b];
+            int **vCol = new int *[w];
+            for (int i = 0; i < w; i++)
+                vCol[i] = new int[side_b * side_b];
+
+            //first position
+            int x = xmin;
+            int y = border;
+
+            int ctrInd = 0;
+            for (int kk = 0; kk < k; kk++)
+                minVals[kk] = std::numeric_limits<int>::max();
```

Comment:
Is this function pure an initialization of values?  Because than it might be better simply initializing them with 0 and removing the extra include needed here. I guess that it is the same for all other uses of numeric limits.